zip-0312.rst

```diff
@@ -1,7 +1,552 @@
+- All key share holders are also trusted with the privacy of the transaction,
+  thus a rogue key share holder will be able to break its privacy and unlinkability.
+
+
+Non-requirements
+================
+
+- This ZIP does not support removing the Coordinator role, as described in #[frost-removingcoordinator]_.
+- This ZIP does not prevent key share holders from linking the signing operation to a transaction in the blockchain.
+- Like the FROST specification [#FROST], this ZIP does not specify a key generation
+  procedure; but refer to that specification for guidelines.
+- Network privacy is not in scope for this ZIP, and must be obtained with other
+  tools if desired.
+
+
+Specification
+=============
+
+Algorithms in this section are specified using Python pseudo-code, in the same
+fashion as the FROST specification [#FROST]_.
+
+The types Scalar, Element, and G are defined in #[frost-primeordergroup]_, as
+well as the notation for elliptic-curve arithmetic, which uses the additive
+notation. Note that this notation differs from that used in the Zcash Protocol
+Specification. For example, `G.ScalarMult(P, k)` is used for scalar
+multiplication, where the protocol spec would use :math:`[k] P` with the group
+implied by :math:`P`.
+
+
+Re-randomizable FROST
+---------------------
+
+To add re-randomization to FROST, follow the specification [#FROST]_ with the
+following modifications.
+
+
+Key Generation
+''''''''''''''
+
+While key generation is out of scope for this ZIP and the FROST spec [#FROST]_,
+it needs to be consistent with FROST, see [#frost-tdkg]_ for guidance. The
+spend authorization private key :math:`\mathsf{ask}` [#protocol-spendauthsig]_
+is the particular key that must be used in the context of this ZIP. Note that
+the :math:`\mathsf{ask}` is usually derived from the spending key :math:`\mathsf{sk}`,
+though that is not required. Doing so might require a trusted dealer key generation
+process as detailed in [#frost-tdkg]_ (as opposed to distributed key generation).
+
+
+Randomizer Generation
+'''''''''''''''''''''
+
+A new helper function is defined, which computes :math:`\mathsf{RedDSA.GenRandom}`:
+
+::
+
+  randomizer_generate():
+
+  Inputs:
+  - None
+
+  Outputs: randomizer, a Scalar
+
+  def randomizer_generate():
+    randomizer_input = random_bytes(64)
+    return H3(randomizer_input)
+
+
+Binding Factor Computation
+''''''''''''''''''''''''''
+
+The `compute_binding_factors` function is changed to receive the `randomizer_point`
+as follows: ::
+
+  Inputs:
+  - commitment_list = [(i, hiding_nonce_commitment_i, binding_nonce_commitment_i), ...],
+    a list of commitments issued by each participant, where each element in the list
+    indicates a NonZeroScalar identifier i and two commitment Element values
+    (hiding_nonce_commitment_i, binding_nonce_commitment_i). This list MUST be sorted
+    in ascending order by identifier.
+  - msg, the message to be signed.
+  - randomizer_point, an element in G.
```

Comment:
Should we call this `randomizer_commitment`? It may be an elliptic curve point, or it may be a group element abstraction, but its function is as a commitment to the actual randomizer, a secret scalar

Reply:
Good point (:rofl:), changed in 7bffa04